src/hand_demo_random.cpp

```diff
@@ -0,0 +1,42 @@
+#include <ros/ros.h>
+#include <message_filters/subscriber.h>
+#include <message_filters/time_synchronizer.h>
+
+#include <moveit/move_group_interface/move_group_interface.h>
+
+#include <random>
+
+#include <iostream>
+#include <string>
+#include <fstream>
+
+// Run through a set of random hand poses with a right shadow hand.
+
+using namespace std;
+
+int main(int argc, char** argv){
+	ros::init(argc, argv, "hand_demo");
+	ros::AsyncSpinner spinner(3);
+	spinner.start();
+	ros::NodeHandle nh;
+	ros::NodeHandle pnh("~");
+
+	bool randomize= pnh.param<bool>("random", true);
+
+	ROS_INFO("setting up MGI");
+	moveit::planning_interface::MoveGroupInterface mgi("right_hand");
+
+	while(ros::ok()){
+		mgi.setRandomTarget();
```

Comment:
That does not sound like a good idea to me...
The random target can be in collision (planning will fail later then)
and you end up having no idea where the hand will go/be next.
Together with the calibration problem of THJ4/J5 this will probably result in more collisions